libnetwork/controller.go

```diff
@@ -1147,7 +1174,14 @@ func (c *controller) NewSandbox(containerID string, options ...SandboxOption) (S
 	}
 
 	if sb.osSbox != nil {
-		// Apply operating specific knobs on the load balancer sandbox
+		err := sb.osSbox.InvokeFunc(func() {
+			sb.osSbox.ApplyOSTweaks(sb.oslTypes)
+		})
+
+		if err != nil {
+			logrus.Errorf("Failed to apply performance tuning sysctls the sandbox: %v", err)
+		}
+		// Keep this just so performance is not changed
 		sb.osSbox.ApplyOSTweaks(sb.oslTypes)
```

Comment:
I'm a bit confused why `sb.osSbox.ApplyOSTweaks(sb.oslTypes)` is called twice (using different mechanisms); could explain?

Reply:
This is more or less due to longstanding discussions around inheritance of namespaced vs non-namespaced sysctls: https://lore.kernel.org/lkml/20160221071102.9686.63148.stgit@buzz/

Values set in outside of the namespace do not _always_ propagate, but setting it both in the ns (through `InvokeFunc->nsInvoke->netns.Set`) provides a "safety valve" against the ongoing changes in kernel namespacing.